junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,447 @@
+package org.scalajs.junit.plugin
+
+import scala.language.reflectiveCalls
+
+import org.scalajs.junit._
+
+import scala.reflect.internal.Flags
+import scala.tools.nsc._
+import scala.tools.nsc.plugins.{
+Plugin => NscPlugin, PluginComponent => NscPluginComponent
+}
+
+class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
+  selfPlugin =>
+
+  val name: String = "Scala.js JUnit plugin"
+
+  val components: List[NscPluginComponent] =
+    List(ScalaJSJUnitPluginComponent)
+
+  val description: String = "Adapts JUnit test classes to Scala.js"
+
+  object ScalaJSJUnitPluginComponent extends plugins.PluginComponent
+      with transform.Transform {
+
+    protected def newTransformer(
+        unit: global.CompilationUnit): global.Transformer = {
+      new ScalaJSJUnitPluginTransformer
+    }
+
+    val global: Global = selfPlugin.global
+    val phaseName: String = "junit-inject"
+    val runsAfter: List[String] = List("mixin")
+    override val runsBefore: List[String] = List("jscode")
+
+    class ScalaJSJUnitPluginTransformer extends global.Transformer {
+
+      import global._
+
+      type ScalaJSPlugin = NscPlugin {
+        def PrepInteropComponent: AnyRef {
+          def genExportMember(sym: Symbol): List[Tree]
+          def registerModuleExports(sym: Symbol): Unit
+          def genNamedExport(defSym: Symbol, jsName: String, pos: Position): Unit
+        }
+      }
```

Comment:
Hummm... Reflective typing? Could you not just add annotations on relevant members? This seems much less fragile...

(I have not seen what exactly the plugin does, is it really necessary to run after mixin?)